classes/submission/form/PKPSubmissionSubmitStep1Form.inc.php

```diff
@@ -78,31 +84,57 @@ function fetch($request) {
-				$authorUserGroup = $userGroupDao->getById($authorUserGroupAssignment->getUserGroupId());
-				if ($userGroupDao->userGroupAssignedToStage($authorUserGroup->getId(), WORKFLOW_STAGE_ID_SUBMISSION)) {
-					$userGroupNames[$authorUserGroup->getId()] = $authorUserGroup->getLocalizedName();
-				}
+
+		// List existing user roles
+		$managerUserGroupAssignments = $userGroupAssignmentDao->getByUserId($user->getId(), $this->context->getId(), ROLE_ID_MANAGER);
+		$authorUserGroupAssignments = $userGroupAssignmentDao->getByUserId($user->getId(), $this->context->getId(), ROLE_ID_AUTHOR);
+
+		// List available author roles
+		$availableAuthorUserGroups = $userGroupDao->getUserGroupsByStage($this->context->getId(), WORKFLOW_STAGE_ID_SUBMISSION, ROLE_ID_AUTHOR);
+		while($authorUserGroup = $availableAuthorUserGroups->next()) {
+			if ($authorUserGroup->getPermitSelfRegistration()){
+				$availableUserGroupNames[$authorUserGroup->getId()] = $authorUserGroup->getLocalizedName();
 			}
-			$templateMgr->assign('authorUserGroupOptions', $userGroupNames);
-		} else {
-			// The user doesn't have any author user group assignments.  They should be either a manager.
-			// Add all manager user groups
-			$managerUserGroupAssignments = $userGroupAssignmentDao->getByUserId($user->getId(), $this->context->getId(), ROLE_ID_MANAGER);
-			if($managerUserGroupAssignments) while($managerUserGroupAssignment = $managerUserGroupAssignments->next()) {
+		}
+
+		// Set default group to default author group
+		$defaultGroup = $userGroupDao->getDefaultByRoleId($this->context->getId(), ROLE_ID_AUTHOR);
+		$noExistingRoles = false;
+		$managerGroups = false;
+
+		// If the user has manager roles, add manager roles and available author roles to selection
+		if (!$managerUserGroupAssignments->wasEmpty()) {
+			while($managerUserGroupAssignment = $managerUserGroupAssignments->next()) {
 				$managerUserGroup = $userGroupDao->getById($managerUserGroupAssignment->getUserGroupId());
 				$userGroupNames[$managerUserGroup->getId()] = $managerUserGroup->getLocalizedName();
 			}
+			$managerGroups = join(__('common.listSeparator'), $userGroupNames);
+			$userGroupNames = array_replace($userGroupNames, $availableUserGroupNames);
+
+			// Set default group to default manager group
+			$defaultGroup = $userGroupDao->getDefaultByRoleId($this->context->getId(), ROLE_ID_MANAGER);
+
+		// else if the user only has existing author roles, add to selection
+		} else if (!$authorUserGroupAssignments->wasEmpty()) {
+			while($authorUserGroupAssignment = $authorUserGroupAssignments->next()) {
+				$authorUserGroup = $userGroupDao->getById($authorUserGroupAssignment->getUserGroupId());
+				$userGroupNames[$authorUserGroup->getId()] = $authorUserGroup->getLocalizedName();
+			}
 
-			$templateMgr->assign('authorUserGroupOptions', $userGroupNames);
+		// else the user has no roles, only add available author roles to selection
```

Comment:
Ah, good catch. I'll add a "Make Submisision" menu item for users with _only_ `ROLE_ID_READER`. Otherwise, the only routes in are after registration and the Submissions frontend page.